.github/workflows/pull-request.yaml

```diff
@@ -88,6 +88,7 @@ jobs:
 
   build-go:
     name: Build go binaries (API Server and CLI)
+    needs: [unit-test-cli, test-server]
```

Comment:
Added this dependency to avoid running tests based on build if our unit tests break by some reason.